### PR TITLE
Clang tidy updates thirdparty

### PR DESCRIPTION
**Pull request by hjmjohnson**
Builds upon #111 and adds changes to third party that make building cleaner.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
This builds upon #111 but makes changes to the third party directories to make cleaner builds at the expense of deviating from the upstream codebase.

**Comment by hjmjohnson** [comment]
@malaterre Could these pull requests please be reviewed?

**Comment by malaterre** [comment]
Same comment as for #111.

File: Utilities/gdcmopenjpeg/src/lib/openjp2/j2k.c [hunk 1]
```diff
@@ -6921,8 +6921,9 @@ OPJ_BOOL opj_j2k_setup_encoder(opj_j2k_t *p_j2k,
                                        image->comps[0].h * image->comps[0].prec) /
                                       ((double)parameters->tcp_rates[parameters->tcp_numlayers - 1] * 8 *
                                        image->comps[0].dx * image->comps[0].dy));
-            if (temp_size > (OPJ_FLOAT32)INT_MAX) {
-                parameters->max_cs_size = (OPJ_FLOAT32)INT_MAX;
+            static const OPJ_FLOAT32 max_float_size_allowed = (1 << FLT_MANT_DIG); // LARGEST INTEGER STORABLE IN FLOAT
+            if (temp_size > max_float_size_allowed) {
+                parameters->max_cs_size = max_float_size_allowed;
             } else {
```
Comment (issakomi) [inline Utilities/gdcmopenjpeg/src/lib/openjp2/j2k.c:6927]: FYI, the warning
```
GDCM/Utilities/gdcmopenjpeg/src/lib/openjp2/j2k.c:6925:43: warning: overflow in implicit constant conversion [-Woverflow]
                 parameters->max_cs_size = (OPJ_FLOAT32)INT_MAX;
```
was fixed in _openjpeg_ **2.3.1**:  C-cast (OPJ_FLOAT32) was simply removed

**2.3.0**
```

            if (temp_size > (OPJ_FLOAT32)INT_MAX) {
                parameters->max_cs_size = (OPJ_FLOAT32)INT_MAX;
            } else {
                parameters->max_cs_size = (int) floor(temp_size);
            }
```

**2.3.1**

```
            if (temp_size > INT_MAX) {
                parameters->max_cs_size = INT_MAX;
            } else {
                parameters->max_cs_size = (int) floor(temp_size);
            }
```

Seems to be actual in _openjpeg_ master
https://github.com/uclouvain/openjpeg/blob/0782cd075d7e9678bf1cb448c1f3d048fabe4489/src/lib/openjp2/j2k.c#L7675

**Pull request closed without merging**